### ISSUE TITLE
Launcher fails to fetch `BedrockLauncher.TokenBroker.dll`

### ISSUE THREAD

**Issue by YonatanKra** [BedrockLauncher/BedrockLauncher#313]
Here's the log output:
```
2022-11-04 09:11:29.1890|DEBUG||Version: 1.0.0.0
2022-11-04 09:11:29.2273|DEBUG||Git Repo: https://github.com/BedrockLauncher/BedrockLauncher
2022-11-04 09:11:29.2273|DEBUG||Git Branch: lite
2022-11-04 09:11:29.2273|DEBUG||Git Commit: 3af40bf
2022-11-04 09:11:29.2273|DEBUG||Git Sha: 3af40bff61ca1767d8d0b6d751a0c13c827e8a83
2022-11-04 09:11:29.2308|DEBUG||Application Starting...
2022-11-04 09:11:29.3830|DEBUG||Application Initalization Started!
2022-11-04 09:11:29.3830|DEBUG||Application Initalization Finished!
2022-11-04 09:11:30.1249|DEBUG||Preparing Application...
2022-11-04 09:11:33.3368|DEBUG||Checking for updates
2022-11-04 09:11:35.5960|DEBUG||Current tag: 1.0.0.0
2022-11-04 09:11:35.5960|DEBUG||Latest tag: 2022.10.9.152
2022-11-04 09:11:35.5960|DEBUG||New version available!
2022-11-04 09:11:35.5960|DEBUG||Preparing Application: DONE
2022-11-04 09:11:48.9438|DEBUG||System.DllNotFoundException: Unable to load DLL 'BedrockLauncher.TokenBroker.dll' or one of its dependencies: The specified module could not be found. (0x8007007E)
   at BedrockLauncher.UpdateProcessor.Authentication.AuthenticationTokenHelper.GetTotalWUAccounts()
   at BedrockLauncher.UpdateProcessor.Authentication.AuthenticationManager.GetWUUsers() in D:\a\BedrockLauncher\BedrockLauncher\modules\BedrockLauncher.UpdateProcessor\Authentication\AuthenticationManager.cs:line 28
   at BedrockLauncher.Controls.Setting_InsiderCombobox.ReloadList(Boolean save) in D:\a\BedrockLauncher\BedrockLauncher\BedrockLauncher\Controls\Setting_InsiderCombobox.xaml.cs:line 28
   at BedrockLauncher.Controls.Setting_InsiderCombobox.Grid_Loaded(Object sender, RoutedEventArgs e) in D:\a\BedrockLauncher\BedrockLauncher\BedrockLauncher\Controls\Setting_InsiderCombobox.xaml.cs:line 39
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   at System.Windows.BroadcastEventHelper.BroadcastEvent(DependencyObject root, RoutedEvent routedEvent)
   at System.Windows.BroadcastEventHelper.BroadcastLoadedEvent(Object root)
   at System.Windows.Media.MediaContext.FireLoadedPendingCallbacks()
   at System.Windows.Media.MediaContext.FireInvokeOnRenderCallbacks()
   at System.Windows.Media.MediaContext.RenderMessageHandlerCore(Object resizedCompositionTarget)
   at System.Windows.Media.MediaContext.RenderMessageHandler(Object resizedCompositionTarget)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
```

**Comment by dfgshdsfh** [comment]
I'm also having a `BedrockLauncher.TokenBroker.dll` issue and am unable to finish running the launcher. It always fails during the initial setup after clicking "Next" on the "One thing..." screen where save/load data is set. I have tried both fixed folder and portable, no change in the app abruptly aborting.

OS: Windows 10 22H2 (19045.2311), fully updated. This was just installed this month, so there shouldn't be any weird things going on that are because of my activity.
BedrockLauncher version: [Lite Update (2022.10.9.152)](https://github.com/BedrockLauncher/BedrockLauncher/releases/tag/2022.10.9.152)
I tried installing each of the following .NET Desktop Runtimes, rebooting after each install. All fail the same way.
[6.0.11 (x64)](https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-desktop-6.0.11-windows-x64-installer)
[6.0.11 (x86)](https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-desktop-6.0.11-windows-x86-installer)
[7.0.0 (x64)](https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-desktop-7.0.0-windows-x64-installer)
Current MC Bedrock version installed through the Microsoft Store: 1.19.50

Log_2022-12-01_23-17-19.log
```
2022-12-01 23:17:19.2421|DEBUG||Version: 1.0.0.0
2022-12-01 23:17:19.2683|DEBUG||Git Repo: https://github.com/BedrockLauncher/BedrockLauncher
2022-12-01 23:17:19.2683|DEBUG||Git Branch: lite
2022-12-01 23:17:19.2683|DEBUG||Git Commit: 3af40bf
2022-12-01 23:17:19.2683|DEBUG||Git Sha: 3af40bff61ca1767d8d0b6d751a0c13c827e8a83
2022-12-01 23:17:19.2683|DEBUG||Application Starting...
2022-12-01 23:17:19.5181|DEBUG||Application Initalization Started!
2022-12-01 23:17:19.5181|DEBUG||Application Initalization Finished!
2022-12-01 23:17:20.9407|DEBUG||Preparing Application...
2022-12-01 23:17:22.7924|DEBUG||Checking for updates
2022-12-01 23:17:23.5880|DEBUG||Current tag: 1.0.0.0
2022-12-01 23:17:23.5880|DEBUG||Latest tag: 2022.10.9.152
2022-12-01 23:17:23.5880|DEBUG||New version available!
2022-12-01 23:17:23.5880|DEBUG||Preparing Application: DONE
2022-12-01 23:17:36.5017|DEBUG||System.DllNotFoundException: Unable to load DLL 'BedrockLauncher.TokenBroker.dll' or one of its dependencies: The specified module could not be found. (0x8007007E)
   at BedrockLauncher.UpdateProcessor.Authentication.AuthenticationTokenHelper.GetTotalWUAccounts()
   at BedrockLauncher.UpdateProcessor.Authentication.AuthenticationManager.GetWUUsers() in D:\a\BedrockLauncher\BedrockLauncher\modules\BedrockLauncher.UpdateProcessor\Authentication\AuthenticationManager.cs:line 28
   at BedrockLauncher.Controls.Setting_InsiderCombobox.ReloadList(Boolean save) in D:\a\BedrockLauncher\BedrockLauncher\BedrockLauncher\Controls\Setting_InsiderCombobox.xaml.cs:line 28
   at BedrockLauncher.Controls.Setting_InsiderCombobox.Grid_Loaded(Object sender, RoutedEventArgs e) in D:\a\BedrockLauncher\BedrockLauncher\BedrockLauncher\Controls\Setting_InsiderCombobox.xaml.cs:line 39
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   at System.Windows.BroadcastEventHelper.BroadcastEvent(DependencyObject root, RoutedEvent routedEvent)
   at System.Windows.BroadcastEventHelper.BroadcastLoadedEvent(Object root)
   at System.Windows.Media.MediaContext.FireLoadedPendingCallbacks()
   at System.Windows.Media.MediaContext.FireInvokeOnRenderCallbacks()
   at System.Windows.Media.MediaContext.RenderMessageHandlerCore(Object resizedCompositionTarget)
   at System.Windows.Media.MediaContext.RenderMessageHandler(Object resizedCompositionTarget)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
```

**Comment by Rayth** [comment]
try Installing from the store, run it first, then uninstall it and try from launcher. Also make sure you're logged into the store and the xbox app

**Comment by IllsonR6** [comment]
> try Installing from the store, run it first, then uninstall it and try from launcher. Also make sure you're logged into the store and the xbox app

This is very strange, everything works fine on the computer, but on the laptop, even after reinstalling Windows, nothing changes

**Comment by IllsonR6** [comment]
> try Installing from the store, run it first, then uninstall it and try from launcher. Also make sure you're logged into the store and the xbox app

Wait, this program only works if you have a license?

**Comment by tresabhi** [comment]
You have to legally own a copy of Minecraft.

**Comment by dfgshdsfh** [comment]
> solved the problem?

The server I was trying to access updated, so I was fine with the normal client for a while. Just now I retried the bedrock launcher, I changed nothing between then an now, I didn't even do anything in regards to sign-ins. Literally just opened the folder I already had tried to use and started the program. This time I was allowed to select my microsoft account, and am now able to use the bedrock launcher to switch versions just fine.

I'm not sure what is different between now and then, but my heaviest suspicion is that it's something to do with sign-ins and microsoft creating a hydra of possible states to be in, one of which the bedrock launcher didn't recognize. Unfortunately for future me or someone else, this is beyond my technical ability to investigate.
